dzBook

```diff
@@ -0,0 +1,110 @@
+#version 1
+
+class Book:
+    
+    
+    def __init__(self):
+        self.book = set()
+    
+    
+    def stats(self):
+        return len(self.book)
+    
+    
+    def show(self, name:str):
+        for word in self.book:
+            if word[0] == name:
+                return word
+    
+    
+    def list(self):
+        return sorted([i[0] for i in self.book])
+    
+    
+    def add(self, name:str, number:int):
+        for word in self.book:
+            if word[0] == name:
+                self.book.remove(word)
+                self.book.add((name, number))
+                break
+        else:
+            self.book.add((name, number))
+    
+    
+    def delete(self, name:str):
+        for word in self.book:
+            if word[0] == name:
+                self.book.remove(word)
+                break
+
+book = Book()
+
+book.add('data', 84930238)
+book.delete('data')
+book.add('date', 53838487)
+
+print(book.show('date'))
+print(book.list())
+print(book.stats())
+
+
+
+#version 2 через бесконечный цикл и ввод команд напрямую в консоль
+
+
+class Book:
+    
+    
+    def __init__(self):
+        self.book = set()
+    
+    
+    def stats(self):
+        print(len(self.book))
+    
+    
+    def show(self):
+        name = input("Введіть ім'я:\n")
+        for word in self.book:
+            if word[0] == name:
+                print(word)
+    
+    
+    def list(self):
+        print(sorted([i[0] for i in self.book]))
+    
+    
+    def add(self):
+        name = input("Введіть ім'я:\n")
+        number = int(input("Введіть номер:\n"))
+        for word in self.book:
+            if word[0] == name:
+                self.book.remove(word)
```

Comment:
According to the task values in phonebook cannot be overridden. User shouldn't be able to add a record if the record already exists. User should remove existing record first, and then add a new one